manifold/balance.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
+
+
+from itertools import chain
+from typing import Literal
+
+from pysad.utils import hex_to_bytes
+
+from manifold.call import Call
+from manifold.constants import (
+    BCHECKER_ADDRESSES,
+    ERC20_BALANCE_SIGNATURE,
+    NATIVE_ADDRESS,
+    NATIVE_BALANCE_SIGNATURE,
+    ZERO_ADDRESS,
+    Network,
+)
+from manifold.multicall import MultiCall
+from manifold.utils import batch
+
+
+class BalanceRequest:
+    token_address: bytes
+    owner_address: bytes
+
+    def __init__(self, token_address: str | bytes, owner_address: str | bytes) -> None:
+        self.token_address = hex_to_bytes(token_address)
+        self.owner_address = (
+            _address
+            if (_address := hex_to_bytes(owner_address)) != ZERO_ADDRESS
+            else NATIVE_ADDRESS
+        )
+
+    def is_native(self) -> bool:
+        return self.owner_address == NATIVE_ADDRESS
```

Comment:
Similar to above, shouldn't this check the token_address instead (+ check if it's 0x)?